### PR TITLE
Porting devhelp plugin for gtk3. Working display in sidebar and main …

### PR DESCRIPTION
**Pull request by liomarhora**
Porting devhelp plugin for gtk3. Working display in sidebar and main window. Missing showing SearchBar in sidebar or other tweaks.

### CODE REVIEW

**Comment by elextr** [comment]
Fails on Linux.  Please remove all build artifacts from the PR.

**Comment by liomarhora** [comment]
> Fails on Linux. Please remove all build artifacts from the PR.

What would that be?

**Comment by elextr** [comment]
automate.cache for starters, also m4 files.

File: MAINTAINERS
Comment (elextr) [inline MAINTAINERS:65]: Have you got Matthew's agreement to be maintainer?

**Comment by elextr** [comment]
What have you done to move the plugin to a supported version of webkit?  It was deleted from geany-plugins because it used a version of webkit that was not supported by most distributions and nobody wanted to port it to the new version because IIUC the API changed significantly.

**Comment by elextr** [comment]
Neither Linux nor Windows CI passes, that needs to be fixed.

**Comment by elextr** [comment]
Why are you building libdevhelp from a static source instead of using the libdevhelp distributed by the distro that is supported with security updates bugfixes etc?

**Comment by liomarhora** [comment]
> Por que você está construindo o libdevhelp a partir de uma fonte estática em vez de usar o libdevhelp distribuído pela distro que é compatível com atualizações de segurança, correções de bugs, etc?



> Why are you building libdevhelp from a static source instead of using the libdevhelp distributed by the distro that is supported with security updates bugfixes etc?

apologies for the delay.
I had an unforeseen event and could only respond now.
I am building with the library provided by the system (Debian).
But where are you checking the static font in the code?

**Comment by liomarhora** [comment]
I accidentally closed the pull.

**Comment by liomarhora** [comment]
> Neither Linux nor Windows CI passes, that needs to be fixed.

How do I perform these tests before pulling?
Why is it working normally on debian and manjaro.

**Comment by elextr** [comment]
The CI runs on Ubuntu 20.04

**Comment by elextr** [comment]
If you are using the library then remove the devhelp sources from the pull.

**Comment by liomarhora** [comment]
> If you are using the library then remove the devhelp sources from the pull.

ok

File: MAINTAINERS
Comment (elextr) [reply MAINTAINERS:65]: Then its rude to put him as maintainer, nominate yourself.  If you are not willing to support the plugin you are proposing it will be less likely to be accepted back.

Comment (liomarhora) [reply MAINTAINERS:65]: Can I put my name and email in place of theirs and their authorization?

Comment (elextr) [reply MAINTAINERS:65]: To be clear, the plugin has been removed from Geany-plugins because it was not usable as it was and broke the build. You are effectively submitting a new plugin that is now usable, not Matthew.  

He and any other authors retain copyright to the code that is reused, you can't change that, but it is licensed under the GPL allowing you to reuse it as you are retaining the GPL.  That is the authorisation.  What the maintainers file says has no impact on that. 

Preferably you should use Git to show the changes from a position when the plugin was in the repository so the copyright is traceable.  I am not a git expert so I have not considered exactly what that would entail, likely reverting the removal commits, then applying your changes on top of that.

The maintainers file is a list of contacts for the person who is willing to provide support (address issues, fix bugs, maybe add features) simply to make it easier for users to find such support.  In many cases that is not the author.  To avoid "drive by" plugin submissions they normally are not accepted unless the submitter is willing to stand behind their work (at least for a while).

**Comment by liomarhora** [comment]
> The CI runs on Ubuntu 20.04
> 
> [Edit: the dependency you ask for is the old version of webkit that is no longer available, so the CI doesn't even start]

I believe I found the place where I requested the old version of webkit.

**Comment by liomarhora** [comment]
I can't find where this libwebkitgtk dependency is in the code

**Comment by elextr** [comment]
Nup, you still tell CI to load libwebkitgtk-dev which does not exist any more.  

Its not in the code its in the CI setup, `build.yml`, the CI hasn't even gotten to compiling your code yet.

The Windows build doesn't fail (it uses a docker image with pre-installed deps), but devhelp does not configure, possibly because that image doesn't have the requisite dependencies.

Maybe because of webkit, @eht16 is there a even a webkit for windows?  Webhelper doesn't configure on windows either.

**Comment by elextr** [comment]
PS because you are a new contributor Github requires that we manually enable each CI run, so there can be delays before it runs.

**Comment by liomarhora** [comment]
> Nup, you still tell CI to load libwebkitgtk-dev which does not exist any more.
> 
> Its not in the code its in the CI setup, `build.yml`, the CI hasn't even gotten to compiling your code yet.
> 
> The Windows build doesn't fail (it uses a docker image with pre-installed deps), but devhelp does not configure, possibly because that image doesn't have the requisite dependencies.
> 
> Maybe because of webkitgtk, @eht16 is there a even a webkitgtk for windows? Webhelper doesn't configure on windows either.

how do i solve this?

**Comment by elextr** [comment]
> Its not in the code its in the CI setup, build.yml, the CI hasn't even gotten to compiling your code yet.

That by editing your changes to `build.yaml` to ask for the correct version.

Windows, forget it, to answer my own question to @eht16 above, google can't find webkitgtk for windows, so it probably doesn't exist.

**Comment by liomarhora** [comment]
> > Its not in the code its in the CI setup, build.yml, the CI hasn't even gotten to compiling your code yet.
> 
> That by editing your changes to `build.yaml` to ask for the correct version.
> 
> Windows, forget it, to answer my own question to @eht16 above, google can't find webkitgtk for windows, so it probably doesn't exist.

I just edited the settings in the devhelp.m4 file.
Was it this file?

**Comment by elextr** [comment]
(/me slips into grumpy olde guy mode)  How is `build.yaml` `devhelp.m4`?

For me `build.yaml` comes up as the very first change on the "Files Changed" tab of the PR on github.

**Comment by liomarhora** [comment]
> (/me slips into grumpy olde guy mode) How is `build.yaml` `devhelp.m4`?
> 
> For me `build.yaml` comes up as the very first change on the "Files Changed" tab of the PR on github.

sorry,
I will try to identify

**Comment by eht16** [comment]
>The Windows build doesn't fail (it uses a docker image with
>pre-installed deps), but devhelp does not configure, possibly because
>that image doesn't have the requisite dependencies.

The Docker image is not really relevant here, it only has some
dependencies as some sort of cache. The workflow definition in the
project is the authority for installing dependecies.

>Maybe because of webkit, @eht16 is there a even a webkit for windows?
>Webhelper doesn't configure on windows either.

There is no more WebkitGTK for Windows:
***@***.***/thread/4DEVGI2ICZX7F7GE6FLSNNQF3Q6I2OP7/

The link in the post is broken, the correct one should be
https://github.com/MSYS2/MINGW-packages/issues/4318.

So, the devhelp plugin won't be available on Windows at all.

**Comment by elextr** [comment]
ok, since the windows CI properly configures devhelp disabled there is nothing to do there.

So @liomarhora [this](.github/workflows/build.yml) just needs to depend on a current package, possibly libwebkit2gtk-4.0-dev (what is installed here on old Mint).  Then the CI should get to your code, as its working that should go smoothly :-).

**Comment by liomarhora** [comment]
> ok, since the windows CI properly configures devhelp disabled there is nothing to do there.
> 
> So @liomarhora [this](.github/workflows/build.yml) just needs to depend on a current package, possibly libwebkit2gtk-4.0-dev (what is installed here on old Mint). Then the CI should get to your code, as its working that should go smoothly :-).

updated version of webkitgtk library in CI.
Could you run the test?

**Comment by elextr** [comment]
Ahhh, now both Linux and Windows break, the `devhelp.m4` should explicitly disable on windows since the libwebkit2 is available on the cross compile host platform and that is what configure tests.   I am m4 illiterate, maybe some expert can advise how.

@liomarhora Linux fail means you have a dependency on something else which you happen to have installed on your machine but which isn't on the CI machine, no idea what.

**Comment by liomarhora** [comment]
> Ahhh, now both Linux and Windows break, the `devhelp.m4` should explicitly disable on windows since the libwebkit2 is available on the cross compile host platform and that is what configure tests. I am m4 illiterate, maybe some expert can advise how.
> 
> @liomarhora Linux fail means you have a dependency on something else which you happen to have installed on your machine but which isn't on the CI machine, no idea what.

I will check

**Comment by liomarhora** [comment]
Is there a way to run the app on the CI of my fork before pulling it on the master?

**Comment by elextr** [comment]
> Is there a way to run the app on the CI of my fork before pulling it on the master?

Possibly, but I don't know how, possibly a github expert can tell what needs to be enabled.

**Comment by liomarhora** [comment]
> > Existe uma maneira de executar o aplicativo no CI do meu garfo antes de puxá-lo no mestre?
> 
> Possivelmente, mas não sei como, possivelmente um especialista do github pode dizer o que precisa ser ativado.

I'll search

**Comment by liomarhora** [comment]
> > Is there a way to run the app on the CI of my fork before pulling it on the master?
> 
> Possibly, but I don't know how, possibly a github expert can tell what needs to be enabled.

**Comment by liomarhora** [comment]
> > Is there a way to run the app on the CI of my fork before pulling it on the master?
> 
> Possibly, but I don't know how, possibly a github expert can tell what needs to be enabled.

hello,
I created a repository and did the tests in the CI and managed to fix the problem.
I've updated the current repository and I'm waiting for it to run on CI.
Could you perform this action for me?

**Comment by liomarhora** [comment]
there was in the POTFILES.in file the path of the old devhelp library. But I already made the changes.

**Comment by liomarhora** [comment]
Hello,
Could take action now

**Comment by liomarhora** [comment]
what is the next step?

**Comment by elextr** [comment]
You need to remove all the product files that have been committed.  Anything that matches the patterns in geany-plugins `.gitignore`.

**Comment by liomarhora** [comment]
> You need to remove all the product files that have been committed. Anything that matches the patterns in geany-plugins `.gitignore`.

I was kind of lost on how to do this removal.
Would you help me?

**Comment by elextr** [comment]
Manually with `git-rm`.  Then do __not__ build until after committing the removal.

**Comment by liomarhora** [comment]
> manualmente com `git-rm`. Então **não** construa até depois de confirmar a remoção.

Before confirming I used the command git clean -fxd to remove the compiled files.
Analyzing the directory I can't find the files that should be excluded.
Could you give me examples of which ones I should exclude.

**Comment by liomarhora** [comment]
there were 23 commits, so I ran git reset --soft HEAD~22 and deleted 96 files.
Was that the procedure?

File: devhelp/AUTHORS
Comment (eht16) [inline devhelp/AUTHORS:1]: Maybe add yourself here?

File: devhelp/src/dhp-plugin.c
Comment (eht16) [inline devhelp/src/dhp-plugin.c:221]: Maybe add yourself here too?

File: devhelp/src/dhp-manpages.c
Comment (eht16) [inline devhelp/src/dhp-manpages.c:1]: Could you add the file header with the authors and license in all source files?

File: devhelp/Makefile.am
Comment (eht16) [inline devhelp/Makefile.am:3]: What's the use of this switch? :D

File: devhelp/autogen.sh
Comment (eht16) [inline devhelp/autogen.sh:1]: The files:
- autogen.sh
- config.status
- configure
- configure.ac
- configure.scan
- libtool
- the whole m4 folder
are not necessary at all, just delete them.

File: devhelp/data/Makefile.am
Comment (eht16) [inline devhelp/data/Makefile.am:8]: `geany-devhelp.png` is missing here or it is not used and can be removed?

File: devhelp/doc/Doxyfile
Comment (eht16) [inline devhelp/doc/Doxyfile:1]: Is this file used anywhere?

**Review by eht16 — CHANGES_REQUESTED** [review]
I had a quick look and left a few comments.
Though this was no proper review, just a first look.

**Comment by eht16** [comment]
> there were 23 commits, so I ran git reset --soft HEAD~22 and deleted 96 files. Was that the procedure?

This removed the commits only for you locally, they are still here in the repository. Just the delete (`git rm` as @elextr said) the files mentioned in one of my comments above.

> hello, I created a repository and did the tests in the CI and managed to fix the problem. I've updated the current repository and I'm waiting for it to run on CI. Could you perform this action for me?

I just approved the CI pipeline, so further pushes will be checked automatically.

File: devhelp/AUTHORS
Comment (liomarhora) [reply devhelp/AUTHORS:1]: I changed it, but Matthew Brush who created the plugin.
I just updated to gtk+3 version

File: devhelp/src/dhp-plugin.c
Comment (liomarhora) [reply devhelp/src/dhp-plugin.c:221]: Updated.

File: devhelp/src/dhp-manpages.c
Comment (liomarhora) [reply devhelp/src/dhp-manpages.c:1]: Yes, done.

File: devhelp/Makefile.am
Comment (liomarhora) [reply devhelp/Makefile.am:3]: The plugin used an old devhelp library. 
That's why this condition exists

**Comment by elextr** [comment]
@liomarhora the problem was that you had committed some of those build files to your branch, perhaps you force added them. 
 As @eht16 said they need to be removed from there, not just the working directory, we don't want your specific build artefacts in the repository.

possibly `git-ls-files -i --exclude-standard` will list them [untried advice]

File: devhelp/autogen.sh
Comment (liomarhora) [reply devhelp/autogen.sh]: ok. done!

File: devhelp/data/Makefile.am
Comment (liomarhora) [reply devhelp/data/Makefile.am:8]: Done.

File: devhelp/doc/Doxyfile
Comment (liomarhora) [reply devhelp/doc/Doxyfile:1]: I still don't know the purpose of this file in the project.
Could you leave it like that?
In the future I will find out about his role in the project.

**Comment by liomarhora** [comment]
> @liomarhorao problema é que você havia confirmado alguns desses arquivos de construção em sua ramificação, talvez você os tenha adicionado à força. Como@eht16disse que eles precisam ser removidos de lá, não apenas do diretório de trabalho, não queremos seus artefatos de construção específicos no repositório.
> 
> possivelmente `git-ls-files -i --exclude-standard`irá listá-los [conselho não experimentado]

Right.
Understood

**Comment by liomarhora** [comment]
I made the modifications.
What is the next step?

File: devhelp/doc/Doxyfile
Comment (eht16) [reply devhelp/doc/Doxyfile:1]: Just delete it.
This is the configuration for Doxygen, a tool to generate API documentation for code. But it is unlikely to be necessary or useful in this context. If you at some point in the future want to use Doxygen, then it is eay to regenerate the config again from scratch.

File: devhelp/Makefile.am
Comment (eht16) [reply devhelp/Makefile.am:3]: Well, I meant that both branches of the switch seem to do the same and so the switch is not necessary at all.

Additionally, Geany-Plugins are always using GTK3, we do not support GTK2 anymore.

File: devhelp/INSTALL
Comment (eht16) [inline devhelp/INSTALL:1]: This file is always not very useful, these are just generic install instructions which mainly don't fit for this plugin.
I'd say better remove it than just keeping it around.

File: devhelp/src/dhp-settings.c
Comment (eht16) [inline devhelp/src/dhp-settings.c:1]: What is this comment about?

File: devhelp/INSTALL
Comment (liomarhora) [reply devhelp/INSTALL]: Done.

File: devhelp/data/Makefile.am
Comment (eht16) [reply devhelp/data/Makefile.am:8]: The file is still not referenced here or am I missing something?

`dist_plugindata_DATA` ensures that files listed here will be included in releases and get installed properly. If you do not add the image filename here, it won't be installed.

File: devhelp/src/dhp-settings.c
Comment (liomarhora) [reply devhelp/src/dhp-settings.c:1]: It was already in this file.
It was Matthew who had put.

**Comment by eht16** [comment]
> I made the modifications. What is the next step?

Be a bit more patient until someone does a full review.
I would like to but cannot say when I find time to do it.

**Comment by liomarhora** [comment]
> > Fiz as modificações. Qual é o próximo passo?
> 
> Seja um pouco mais paciente até que alguém faça uma revisão completa. Eu gostaria, mas não posso dizer quando encontrarei tempo para fazê-lo.

All good.
I'm new with programming.
I thank you for your patience.

**Comment by liomarhora** [comment]
Hi,
While the plugin was not accepted, I'm working on it.
I made some changes regarding Zoom.

**Pull request closed without merging**
